geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentKafkaDataStoreFactory.scala

```diff
@@ -54,14 +62,50 @@ object ConfluentKafkaDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogg
   override val DisplayName = "Confluent Kafka (GeoMesa)"
   override val Description = "Confluent Apache Kafka\u2122 distributed log"
 
-  val SchemaRegistryUrl = new GeoMesaParam[URL]("kafka.schema.registry.url", "URL to a confluent schema registry server, used to read Confluent schemas (experimental)")
+  val SchemaRegistryUrl =
+    new GeoMesaParam[URL](
+      "kafka.schema.registry.url",
+      "URL to a confluent schema registry server, used to read Confluent schemas (experimental)",
+      optional = false
+    )
+
+  val SchemaOverrides =
+    new GeoMesaParam[String](
+      "kafka.schema.overrides",
+      "Typesafe configuration defining a map from topic name to schema override (experimental)",
+      optional = true,
+      largeText = true
+    )
 
   override val ParameterInfo: Array[GeoMesaParam[_ <: AnyRef]] =
-    KafkaDataStoreFactory.ParameterInfo.+:(SchemaRegistryUrl)
+    KafkaDataStoreFactory.ParameterInfo ++ Array(SchemaRegistryUrl, SchemaOverrides)
 
   override def canProcess(params: java.util.Map[String, _ <: java.io.Serializable]): Boolean = {
-    KafkaDataStoreFactoryParams.Brokers.exists(params) &&
-        KafkaDataStoreFactoryParams.Zookeepers.exists(params) &&
-        SchemaRegistryUrl.exists(params)
+    KafkaDataStoreParams.Brokers.exists(params) &&
+      KafkaDataStoreParams.Zookeepers.exists(params) &&
+      SchemaRegistryUrl.exists(params)
+  }
+
+  private[confluent] def parseSchemaOverrides(schemaOverridesConfig: String): Map[String, (SimpleFeatureType, Schema)] = {
+    try {
+      Option(schemaOverridesConfig)
+        .map(ConfigFactory.parseString(_).resolve())
+        .filter(_.hasPath("schemas"))
+        .map {
+          _.getObject("schemas").asScala.map { case (topic, schemaConfig) =>
+            try {
+              val schema = new Schema.Parser().parse(schemaConfig.render(ConfigRenderOptions.concise()))
+              val sft = AvroSimpleFeatureTypeUtils.schemaToSft(schema, Some(topic))
+              topic -> (sft, schema)
+            } catch {
+              case NonFatal(ex) =>
+                throw new IllegalArgumentException(s"Schema override for topic '$topic' is invalid: ${ex.getMessage}")
```

Comment:
generally we try to avoid throwing and immediately re-catching an exception, as it's a somewhat confusing control flow, but it's ok for now.

Reply:
Would it make sense to just remove the outer try-catch?

Reply:
no, i think you'd want it for the `ConfigFactory.parseString` call. possibly you could refactor to use `Try` with `flatMap` and then check for `Success` or `Failure` at the end.